runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -101,7 +101,7 @@ void J9::RecognizedCallTransformer::process_java_lang_StringCoding_encodeASCII(T
    // Note that *latin1FieldAddress will be correct for AOT compilations as well because String.LATIN1 is a final static field with
    // a constant initializer. This means its value is determined by the ROM class of String, and therefore when the String class chain validation succeeds
    // on AOT load, the value is guaranteed to be the same as the one we see during AOT compilation.
-   TR::Node *constNode = TR::Node::iconst(node, fej9->dereferenceStaticFinalAddress(latin1FieldAddress, TR::Int32));
+   TR::Node *constNode = TR::Node::iconst(node, ((TR_J9VM *)fej9)->dereferenceStaticFinalAddress(latin1FieldAddress, TR::Int32).dataInt32Bit);
```

Comment:
Is it guaranteed that we can dereference this field ? i.e. we cannot fail to do so for some reason, e.g. OSR is disabled ? Maybe it is always allowed to dereference this particular field since it is a special known one ?

Reply:
In Java 11 and 17 this field is defined inside the String class:
```
@Native static final byte LATIN1 = 0;
```
Would the fact that this is a static final field guarantee that we can always dereference it?

I can also try testing by disabling OSR to see if we are still able to dereference this field. Just the `-Xjit:disableOSR` option would be sufficient to test this right?

Based on my offline discussions with @jdmpapin and his comment [here](https://github.com/eclipse-openj9/openj9/pull/13855#discussion_r798054085), we shouldn't have a problem for AOT compilations.

Reply:
I talked about AOT in the linked comment, but I just want to add that I don't think this should need any special protection (OSR) to allow constant-folding. We already fold final statics defined in JCL classes without OSR (with the exception of `System`). Additionally, even `javac` will folds uses of this field. And `javac` will even fold uses in other classes. The constant value in the initializer can't be changed without breaking the class's 'binary compatibility.' So I'd say this is especially safe to fold